rosbag2_cpp/include/rosbag2_cpp/writers/sequential_writer.hpp

```diff
@@ -102,13 +106,37 @@ class ROSBAG2_CPP_PUBLIC SequentialWriter
    */
   void write(std::shared_ptr<rosbag2_storage::SerializedBagMessage> message) override;
 
+protected:
+  /**
+   * Initialize the compressor.
+   *
+   * \throws runtime_error If the compression implementation does not exist.
+   */
+  virtual void init_compression(const CompressionOptions & compression_options);
```

Comment:
what happens if the bag is already opened with a set of specific compression options and then this function is called with a different, potentially conflicting, set of it?

Reply:
This function is only called in `open()` and is a protected method (for unit testing) so we don't expect to call it anywhere else.
We could simply move the functionality into `open()` only but that might make the function too large.